tests/virtualization/xen/hotplugging.pm

```diff
@@ -97,5 +98,20 @@ sub run {
     assert_script_run "virsh detach-interface $_ bridge --mac " . $mac{$_} foreach (keys %xen::guests);
 }
 
-1;
+sub run {
+    my ($self) = @_;
+    my $start_run = time();
+    $self->run_hotplugging;
+    my $stop_run = time();
+    return unless get_var("VIRT_AUTOTEST");
+    my $run_time = duration_calc_in_hms($self, $stop_run - $start_run);
+    make_junit_log($self, 'PASSED', $run_time, 'Hotplugging', 'Hotplugging');
+}
+
+sub post_fail_hook {
+    my ($self) = shift;
+    $self->SUPER::post_fail_hook;
+    make_junit_log($self, 'FAILED', 'inf', 'Hotplugging', 'Hotplugging');
+}
 
```

Comment:
IMPORTANT:
The current design which reworks new features tests like snapshot&hotplugging is duplicating. Better to avoid it from class hierarchy.

Reply:
Let's talk face to face for new class hierarchy design this afternoon. Ok?

Reply:
More exactly, it is not duplicating, they just should be handled in the similar way. I am not sure what you want to avoid. You can just say it here directly.

Reply:
More abstraction to avoid the "duplication" in every feature test pm like hotplugging.pm and virsh_external_snapshot.pm and virsh_internal_snapshot.pm for parts like:

```
    my $start_run = time();
    ...
    my $stop_run = time();
    return unless get_var("VIRT_AUTOTEST");
    my $run_time = duration_calc_in_hms($self, $stop_run - $start_run);
```

Reply:
There will be a change to some extent in terms of how $run_time will be calculated. But you can not avoid getting $start_run and $stop_run explicitly, you will have to get, for example, $start_run somewhere, either here or there. You just can not get rid of it.